blog/2024-11-27-flet-v-0-25-release-announcement.md

```diff
@@ -0,0 +1,294 @@
+
+## New packaging
+
+Flet packaging for iOS and Android has been relying on Kivy and it was super annoying when your app depends on Python binary packages, such as Numpy or Pillow. You needed to compile those packages yourself using Kivy command line tools. It was really frustrating and even hopeless if Kivy didn't have "recipes" for some packages, like Pydantic.
+
+Flet does not depend on Kivy anymore and uses its own Python runtime "meticulously crafted in-house".
+
+Flet packaging implementation for iOS and Androind adheres to strict specifications defined in [PEP 730](https://peps.python.org/pep-0730/) (iOS) and [PEP 738](https://peps.python.org/pep-0738/) (Android) which were implemented and released in Python 3.13 (and back-ported to Python 3.12). When pypi.org supports wheel tags for iOS and Android and 3rd-party Python package maintainers start uploading their mobile packages Flet will be compatible with them and you'll be able to use them in your Flet app.
+
+### Pre-built binary packages
+
+`flet build` command for iOS and Android is now installing pre-built binary packages from https://pypi.flet.dev.
+
+New packages can be built with creating a recipe in [Mobile Forge](https://github.com/flet-dev/mobile-forge) project. For now, Flet team is authoring those recipes for you, but when the process is polished and fully-automated you'll be able to send a PR and test the compiled package right away.
+
+If you don't yet see a package you require at https://pypi.flet.dev, you can request it in [Flet discussions - Packages](https://github.com/flet-dev/flet/discussions/categories/packages). Please do not request pure Python packages. Go to package's "Download files" section at https://pypi.org and make sure it contains binary platform-specific wheels.
+
+Packaging behavior was changed too:
+
+- The packaging is not trying to replace `flet` dependency with `flet-runtime`, `flet-embed` or `flet-pyodide`, but install all dependencies "as is" from `requirements.txt` or `pyproject.toml` - thanks to the [new Flet packages structure](#new-python-packages-structure).
+- If the binary package for target platform is not found the packaging won't be trying to compile it from source distribution, but will fail instead with a meaningful error.
+
+### Python 3.12
+
+Packaged Flet app runs on Python 3.12.7 runtime on all platforms.
+
+### Permissions
+
+New `flet build` command allows granular control over permissions, features and entitlements embedded into `AndroidManifest.xml`, `Info.plist` and `.entitlements` files.
+
+No more hard-coded permissions in those files!
+
+For example, setting permissions for iOS bundle:
+
+```
+flet build --info-plist NSLocationWhenInUseUsageDescription=This app uses location service when in use.
+```
+
+or the same in `pyproject.toml` (read about `pyproject.toml` support below):
+
+```toml
+[tool.flet.ios.info] # --info-plist
+NSCameraUsageDescription = "This app uses the camera to ..."
```

Comment:
Camera is not yet implemented and this might lead some users to thinking it is available.
```suggestion
NSLocationWhenInUseUsageDescription = "This app uses location service when in use."
```